### DISCUSSION TITLE
Plan or goal

### DISCUSSION THREAD

**Discussion by wu-clan** [fastapi-practices/fastapi_best_architecture#35] [Ideas]
We can have some discussions here, such as what we want to achieve.

**Comment by wu-clan** [comment]
Hi, @downdawn 

I would like to ask about the work you are doing, the related table？RBAC？

My idea is to copy something we want from gin-vue-admin, although I refer to it in my previous demo work.

**Reply by downdawn** [reply]
Hi, @wu-clan 

I am doing background management of a system now, which may involve multi-tenancy

The groundwork is similar to this project: [https://github.com/go-admin-team/go-admin](https://github.com/go-admin-team/go-admin)

demo：[https://antd.go-admin.pro/](https://antd.go-admin.pro/)

**Reply by wu-clan** [reply]
Will this involve some changes to this warehouse? Or are you just doing your work locally?

**Reply by downdawn** [reply]
Does not conflict with this warehouse, the actual project will be more complicated

What I expect is that this warehouse can be made into a similar basic demo, including front-end and back-end and documentation, which can be used out of the box

**Reply by wu-clan** [reply]
Very good idea

**Comment by wu-clan** [comment]
Hi @downdawn 

What do you think of #36?

Will we use more 3.10 features?

**Reply by downdawn** [reply]
Including the Annotated feature recommended by FastAPI is a built-in feature in higher versions.

I think using a newer stable version is a better choice, taking all factors into consideration.

**Reply by wu-clan** [reply]
Yeah.

I didn‘t expect anyone to use it in py3.8.

I will add some tips to temporarily back to 3.8.

After that, it must be 3.10 or higher.

**Reply by wu-clan** [reply]
Forward compatibility would bring too many manual changes and I decided not to add hints that

Now, I think it is possible to use 3.10+ more features and it must start with 3.10

**Comment by wu-clan** [comment]
- [x] Move the services folder to the root of the app

**Reply by wu-clan** [reply]
Combined with the actual project, this is a fairly uniform design, although I prefer to keep all the functional modules of the api together 

Will be submitted together with the next major update

**Reply by downdawn** [reply]
Okay, should we separate "auth" and "user"?

**Reply by wu-clan** [reply]
auth keeps only the login interface?

**Reply by downdawn** [reply]
login and logout
register is also a big module，may involve verification codes, third-party logins, etc.
user a module

**Reply by wu-clan** [reply]
Even if register requires an authentication code, that may be minimal code or an external wrapper method, and third-party Login should be in the auth module, auth and user is enough, which I think

Also, I do not see the login interface in go-admin?

**Reply by downdawn** [reply]
sys_router.go
[https://github.com/go-admin-team/go-admin/blob/31cd1ee7680267f170a100aada4d84b4401db5cb/app/admin/router/sys_router.go#L71](https://github.com/go-admin-team/go-admin/blob/31cd1ee7680267f170a100aada4d84b4401db5cb/app/admin/router/sys_router.go#L71)

**Reply by wu-clan** [reply]
thanks.

**Comment by wu-clan** [comment]
- [x] Add pycasbin-based RBAC #41

**Comment by wu-clan** [comment]
- [x] token logout #44

**Comment by wu-clan** [comment]
- [x] Custom Encoder #50

**Comment by wu-clan** [comment]
- [x] token refresh #62

**Comment by wu-clan** [comment]
- [x] Update tests #68

**Comment by wu-clan** [comment]
- [x] add rate limit #70

**Comment by downdawn** [comment]
- [x] Add log interface #49

**Comment by wu-clan** [comment]
- [x] Add jwt middlerware #82

**Comment by wu-clan** [comment]
- [x] New token processing logic #83

**Comment by wu-clan** [comment]
- [x] Synchronization function handling #85

**Comment by wu-clan** [comment]
- [x] Add Captcha function #102

**Comment by wu-clan** [comment]
Router retrofit #110

**Reply by wu-clan** [reply]
This solution was abandoned

**Comment by wu-clan** [comment]
- [x] Dictionary management #109

**Comment by wu-clan** [comment]
- [x] RADEME pseudo three-tier architecture description #106

**Comment by wu-clan** [comment]
- [x] Update permission logic #120

**Comment by wu-clan** [comment]
Pause for updates here.
